oeps/oep-0042-bp-authentication.rst

```diff
@@ -0,0 +1,264 @@
+---------------
+
+The currently supported and recommended method of authentication is an OAuth 2.0 implementation using `JSON Web Tokens (JWTs)`_ as OAuth tokens.
+
+Here are a variety of details around this decision:
+
+* Standardize on the `Django OAuth Toolkit (DOT) library`_ to support our OAuth2 implementation.
+
+  * Read the `ADR on Django OAuth Toolkit (DOT)`_.
+
+* JWT Authentication is the standard method of authentication in Open edX.
+
+  * Use JWT and DOT to implement OAuth2. See the `ADR to use JWTs as OAuth Tokens`_ for more details on this decision.
+
+  * JWTs are now the only fully supported type of access token.
+
+  * All JWTs will be signed using "asymmetric" cryptographic keys, as detailed in this `ADR on Asymmetric JWTs`_.
+
+  * For authentication from our micro-frontends, a shared cookie is used to store the JWT and usable across IDAs, as detailed in this `JWT Cookies ADR`_.
+
+Implementation of all the OAuth2/JWT APIs supported by DOT in the LMS Identity Provider can be found in `oauth_dispatch (edx-platform)`_.
+
+* As of the Juniper Open edX release, the deprecated implementation of OAuth2 using OpenID Connect (OIDC) and the `Django OAuth2 Provider (DOP) library`_ has been fully removed. See this `oauth_dispatch as router ADR`_ for additional details of how this transition was implemented in edx-platform, and to better understand the history of ``oauth_dispatch``.
+
+.. _JSON Web Tokens (JWTs): https://tools.ietf.org/html/rfc7519
+.. _Django OAuth Toolkit (DOT) library: https://django-oauth-toolkit.readthedocs.io/en/latest/
+.. _ADR on Django OAuth Toolkit (DOT): https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0002-migrate-to-dot.rst
+.. _ADR to use JWTs as OAuth Tokens: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0003-use-jwt-as-oauth-tokens-remove-openid-connect.rst
+.. _ADR on Asymmetric JWTs: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0008-use-asymmetric-jwts.rst
+.. _JWT Cookies ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0009-jwt-in-session-cookie.rst
+.. _oauth_dispatch (edx-platform): https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/README.rst
+.. _Django OAuth2 Provider (DOP) library: https://django-oauth2-provider.readthedocs.io/en/latest/
+.. _oauth_dispatch as router ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0004-oauth-dispatch-as-router.rst#L33
+
+
+OAuth2 and Bearer Tokens
+------------------------
+
+This section refers to Bearer Tokens as documented in `OAuth 2.0 RFC: Bearer Token Usage`_. These tokens are sent using "Bearer" in the Authorization header field in the request. In contrast, our JWTs are sent using "JWT" in the Authorization header field.
```

Comment:
Suggestion: Add a link for "Authorization header" (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) since it may be confusing to readers why an Authz header is needed for Authn.  As the link explains, this header contains the credentials to **authenticate** a user agent with a server.  This is a wrongly named header, IMO.

Reply:
@nasthagiri: FYI: I updated this.

Reply:
Thanks!